presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -937,6 +937,26 @@ public void testCastNullToColumnTypes()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testCreateEmptyNonBucketedPartition()
+    {
+        String tableName = "test_insert_empty_partitioned_unbucketed_table";
```

Comment:
Does this test the scenario of `STAGE_AND_MOVE_TO_TARGET_DIRECTORY`? This might sound noob but I didn't see where the write mode is set.

Reply:
I think it doesn't matter. For empty partition, we just need to create an empty directory and we can directly create it as the target path.

Reply:
@shixuan-fan :  `STAGE_AND_MOVE_TO_TARGET_DIRECTORY` for open source HDFS, unless underlying file system is S3 in which rename is expensive . See https://github.com/prestodb/presto/blob/2eafd8681fe78a931e5f61af3b95d610a16b88fc/presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java#L63-L66

and 

https://github.com/prestodb/presto/blob/2eafd8681fe78a931e5f61af3b95d610a16b88fc/presto-hive/src/main/java/com/facebook/presto/hive/HiveLocationService.java#L87-L92

The session property `hive.temporary-staging-directory-enabled` was added recently by https://github.com/prestodb/presto/pull/12199 for S3 filesystem in disguise.